syntax = "proto2";

package msg;

message RenderRequest
{
	required string 	message_name = 1;
	
	required string 	asset_uuid = 2;
}

message AssetResponse
{
	required string 	message_name = 1;
	
	required string 	asset_uuid = 2;
	
	required string		asset_type = 3;
	
	required bytes 		asset_content = 4;
	
	required uint32 	frame_count = 5;
	
	required uint32		fps = 6;
}

message FrameResponse
{
	required string 	message_name = 1;
	
	required string 	asset_uuid = 2;
	
	message	PerFrameInfo
	{
		message PerVertexInfo
		{
			required uint32	vertex_index = 1;
			
			required float	x = 2;
			
			required float	y = 3;
			
			required float	z = 4;
		}
		
		required uint32			frame_index = 1;
		
		repeated PerVertexInfo	vertex_list = 2;
	}
	
	repeated PerFrameInfo	frame_list = 3;
}